### PR TITLE
Spark: Introduce companion objects for DELETE and MERGE rewrite rules

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR introduces companion objects for DELETE and MERGE rewrite rules and moves constants there.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
cc @dilipbiswal @rdblue

File: spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDelete.scala [hunk 1]
```diff
@@ -42,13 +42,15 @@ import org.apache.spark.sql.catalyst.utils.RewriteRowLevelOperationHelper
...
+import org.apache.spark.sql.execution.datasources.v2.ExtendedDataSourceV2Implicits
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types.BooleanType
 
 // TODO: should be part of early scan push down after the delete condition is optimized
 case class RewriteDelete(spark: SparkSession) extends Rule[LogicalPlan] with RewriteRowLevelOperationHelper {
 
-  import org.apache.spark.sql.execution.datasources.v2.ExtendedDataSourceV2Implicits._
+  import ExtendedDataSourceV2Implicits._
+  import RewriteRowLevelOperationHelper._
```
Comment (rdblue) [inline spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDelete.scala:53]: Does it work if this import is done in `RewriteRowLevelOperationHelper`?

**Review by rdblue — APPROVED** [review]

File: spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDelete.scala (hunk 1 above)
Comment (aokolnychyi) [reply spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDelete.scala:53]: Unfortunately, no :(

**Comment by dilipbiswal** [comment]
LGTM

**Comment by aokolnychyi** [comment]
Thanks, @rdblue and @dilipbiswal!

**Pull request merged**